cmd/buildah/manifest.go

```diff
@@ -307,7 +307,15 @@ func manifestAddCmd(c *cobra.Command, args []string, opts manifestAddOpts) error
 
 	digest, err := list.Add(getContext(), systemContext, ref, opts.all)
 	if err != nil {
-		return err
+		var storeErr error
+		// check if the local image exists
+		if ref, _, storeErr = util.FindImage(store, "", systemContext, imageSpec); storeErr != nil {
```

Comment:
The local image has been checked above in line297?
 https://github.com/containers/buildah/pull/2955/files#diff-f87fb1de87ea57699bf6ddd7408c420b8bd7daa0f4ef41639120453a94c52b14R297

Reply:
That check never happens since the imagespec is parsed correctly as a remote image, so we always fail if we use a local image that does not exists at a registry.

Reply:
The remote image is never checked until this call.

Reply:
> That check never happens since the imagespec is parsed correctly as a remote image, so we always fail if we use a local image that does not exists at a registry.

LGTM. We could also delete check L296~L297 since they can't be reached.

Reply:
No I can reach there by attempting

./bin/buildah manifest add myimage  sha256:c1876ff0a66a49ae90dbe6ac1d8c413107fec317987c4e2059cf72b430f5a023

Which is also valid.